scripts/sync_gap_analysis_table.py

```
@@ -0,0 +1,78 @@
+
+
+def _normalize_pg_url(url: str) -> str:
+    if url.startswith("postgres://"):
+        return "postgresql://" + url[len("postgres://") :]
+    return url
+
+
+def _pg_host_is_loopback(url: str) -> bool:
+    p = urllib.parse.urlparse(_normalize_pg_url(url))
+    h = (p.hostname or "").lower()
+    return h in ("127.0.0.1", "localhost", "::1", "0.0.0.0") or h == ""
```

Ruff (0.15.15)

[error] Possible binding to all interfaces

(S104)